hal/test-headers/us_ticker_api_tests.h

```diff
@@ -0,0 +1,63 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017-2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** \addtogroup hal_specification */
+/** @{*/
+
+#ifndef MBED_US_TICKER_API_TESTS_H
+#define MBED_US_TICKER_API_TESTS_H
+
+#include "device.h"
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * \defgroup hal_specification_us_ticker Microsecond Ticker
+ * Microsecond ticker specification and tests
+ *
+ * To run the microsecond ticker hal tests use the command:
+ *
+ *     mbed test -t <toolchain> -m <target> -n tests-mbed_hal-us_lp_ticker*,tests-mbed_hal-us_ticker*
+ *
+ * # Defined behavior
```

Comment:
Would it be possible to have these informations in us_ticker_api.h ?

Reply:
@pan- Yeah, I think that would be good. I have already made this change to the RTC specification in #5008. If you get a chance, take a look and let me know what you think.

Also, the bulk of the defined behavior is located in ticker_api_tests.h at the moment. I'm thinking I might move this into us_ticker_api_tests.h and get rid of the additional test header files. Then the lp ticker could reference the us ticker for the common behavior. What do you think of this?